chall03/osalmine.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"log"
+	"strings"
+	"strconv"
+)
+
+func get_rgb_id (s string) (int, int, int, int) {
+	str := strings.Split(s, " ")
+	vals := strings.Split(str[0], ",")
+	id, err := strconv.Atoi(strings.Split(vals[0], "=")[1]);
+	r, err := strconv.Atoi(strings.Split(vals[1], "=")[1]);
+	g, err := strconv.Atoi(strings.Split(vals[2], "=")[1]);
+	b, err := strconv.Atoi(strings.Split(vals[3], "=")[1]);
+	if err != nil {
+		log.Fatal(err)
+	}
+	return id, r, g, b
+}
+
+func main() {
+	url := "https://chall03.hive.fi/"
+	resp, err := http.Get(url)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	s := fmt.Sprintf("%s", body)
```

Comment:
Can you explain why you did that? I don't really understand the purpose

Reply:
If you mean `s := fmt.Sprintf("%s", body)`, I was just converting the body that was a list of decimals I believe into a string. The only way I knew how to do this was to use sprintf. There is probably a better way of doing this but this was my first go code 😅

Reply:
You did a pretty good job.

I think `s := string(body)` is what you are looking for.